File: onlinejudge/programmers/1/카드 뭉치.java
class Solution {
    public String solution(String[] cards1, String[] cards2, String[] goal) {
        int cards1Index = 0;
        int cards2Index = 0;
        
        for (String word : goal) {
            if (cards1Index <= cards1.length - 1 && cards1[cards1Index].equals(word)) {
                cards1Index += 1;
                continue;
            }
            else if (cards2Index <= cards2.length - 1 && cards2[cards2Index].equals(word)) {
                cards2Index += 1;
                continue;
            } else {
                return "No";
            }
        }
        
        return "Yes";
    }
}

File: onlinejudge/programmers/1/카드 뭉치.go
func solution(cards1 []string, cards2 []string, goal []string) string {
    
	cards1Index := 0
	cards2Index := 0

	for _, word := range goal {
		if cards1Index < len(cards1) && cards1[cards1Index] == word {
			cards1Index++
			continue
		} else if cards2Index < len(cards2) && cards2[cards2Index] == word {
			cards2Index++
			continue
		} else {
			return "No"
		}
	}

	return "Yes"
}
